server/application-server/src/main/java/de/tum/in/www1/hephaestus/leaderboard/LeaderboardService.java

```diff
@@ -36,14 +37,17 @@ public LeaderboardService(UserService userService) {
         this.userService = userService;
     }
 
-    public List<LeaderboardEntry> createLeaderboard() {
+    public List<LeaderboardEntry> createLeaderboard(Optional<OffsetDateTime> before, Optional<OffsetDateTime> after) {
         logger.info("Creating leaderboard dataset");
 
         List<User> users = userService.getAllUsers();
-        logger.info("Leaderboard has " + users.size() + " users");
 
-        OffsetDateTime cutOffTime = new Date(System.currentTimeMillis() - 1000 * 60 * 60 * 24 * timeframe)
-                .toInstant().atOffset(ZoneOffset.UTC);
+        OffsetDateTime afterCutOff = after.isPresent() ? after.get()
+                : new Date(System.currentTimeMillis() - 1000 * 60 * 60 * 24 * timeframe).toInstant()
```

Comment:
Maybe we should round to the beginning of the day?

Reply:
Or what happens with `2024-09-10`? I guess it will use midnight, then it might be important for the `after` case. If you for example use `2024-09-15` as `before` I think it should get all timestamps before the end of day. So actually we would need to round up?

Reply:
If no timestamp is provided, it should default to midnight, yes.

While I personally find different default timestamps for `before` and `after` confusing, I do see why this would be expected behavior for users. Will change it 👍

Reply:
Had a look through Java best practices and found this sheet: https://i.sstatic.net/WyAl2.png

Equivalently to your idea, before and after are now only dates (`LocalDate` - without timestamp) with with an interpretation as the start and end of the day on the server respectively. Also looks much cleaner in the URL (example: http://localhost:4200/?after=2024-09-21&before=2024-09-23).

Implemented this change in b76ed6d.